### PR TITLE
M2.23 update login to v3

### PR DESCRIPTION
**Pull request by eartavia-trilogy**
https://github.com/trilogy-group/cloudfix-aws/issues/3591

login endpoint was already at v3

* fix README
* fix panic when logging in
* improve logging

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -20,14 +20,7 @@ This is a Golang [cobra](https://github.com/spf13/cobra/blob/main/user_guide.md)
 
 - Testing with bats
 
-    1. Download bats files (if not done already). Run commands
```
Comment (eartavia-trilogy) [inline README.md:23]: This was outdated

File: cdk/orchestrator.go [hunk 2]
```diff
@@ -105,7 +105,7 @@ func (o *Orchestrator) reco(recoProfile string, workspaceDir string,
 	cloudfixResponses, err := cloudfixManager.GetRecos()
 	if err != nil {
 		dlog.Error("Failed to fetch recommendation from Cloudfix: ", err)
-		logger.Error("Failed to fetch recommendation from Cloudfix")
+		logger.Error("Failed to fetch recommendation from Cloudfix: ", err)
```
Comment (eartavia-trilogy) [inline cdk/orchestrator.go:108]: The log shown in the command line didn't say what the failure was when the credentials were wrong

File: cloudfixIntegration/cloudfixAuth.go [hunk 3]
```diff
@@ -223,9 +223,6 @@ func (ca *CloudfixAuth) validate(token string) (int, error) {
 	if err != nil {
 		return 0, &customError{GenericError, "Failed to form request for accounts endpoint"}
 	}
-	defer func(Body io.ReadCloser) {
```
Comment (eartavia-trilogy) [inline cloudfixIntegration/cloudfixAuth.go:226]: This was causing a panic and [the docs](https://pkg.go.dev/net/http#Request) say it's not required:
image

File: .gitignore [hunk 4]
```diff
@@ -16,7 +16,6 @@ cloudfix-linter-public.exe
 cdklogs/*
 .history
 settings.json
-cdk.json
```
Comment (eartavia-trilogy) [inline .gitignore:19]: This file is actually required for the demo to work.

**Review by flucchese — APPROVED** [review]
LGTM
I can see that globalConfigs.go already points to v3 endpoints.

**Pull request merged**